### PR TITLE
fix(project): Fix klaatu version creation task.

### LINKED ISSUE #13551: Klaatu tests against incorrect versions
It looks like using the versions build into experimenter are not always reliable as there are some versions that may not exist. Lets use the whattrainisitnow URL to get accurate release versions instead.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/EXP-5968)

### PR DESCRIPTION
**Pull request by b4handjr**
Because

- When testing klaatu, the versions it created were not correct and some did not exist.

This commit

- Updates the algorithm to create the versions correctly, ensuring that for each major version, we get the last release version of it.

Fixes #13551

### CODE REVIEW

File: experimenter/experimenter/klaatu/tasks.py [hunk 1]
```diff
@@ -67,41 +67,48 @@ def get_workflows(application: str) -> list[str]:
...
-    for v in versions:
-        parsed_version = version.parse(v)
-        major = parsed_version.release[0]  # type: ignore
-
-        if major not in latest_per_major or parsed_version > latest_per_major[major]:
-            latest_per_major[major] = parsed_version
-
-    final_versions.extend([str(v) for v in latest_per_major.values()])
+    # get list of all releases within experiment range
+    for item in releases:
```
Comment (jaredlockhart) [inline experimenter/experimenter/klaatu/tasks.py]: `for release in releases`

File: experimenter/experimenter/klaatu/tasks.py [hunk 2]
```diff
@@ -67,41 +67,48 @@ def get_workflows(application: str) -> list[str]:
...
+            version.parse(item) >= version.parse(experiment.firefox_min_version)
+            and version.parse(item) <= max_version
+        ):
+            major_list.append(item)
+
+    # get just the major release version number, i.e.: 140
+    for release in major_list:
+        major_versions.add(version.parse(release).major)
+
+    for item in major_versions:
```
Comment (jaredlockhart) [inline experimenter/experimenter/klaatu/tasks.py]: `for major_version in major_versions`

**Comment by b4handjr** [comment]
Fixed some edge cases and added another test. Ran it locally talking to klaatu and everything checks out. https://github.com/mozilla/klaatu/actions/runs/17929207964
